featuretools/tests/computational_backend/test_pandas_backend.py

```diff
@@ -488,6 +488,79 @@ def test_agg_empty_child(es, backend):
     assert df["COUNT(log)"].iloc[0] == 0
 
 
+def test_diamond_entityset(diamond_es):
+    es = diamond_es
+
+    def get_relationship(child, parent):
+        return next(r for r in es.relationships
+                    if r.child_entity.id == child and r.parent_entity.id == parent)
+
+    c_to_r = get_relationship('customers', 'regions')
+    t_to_c = get_relationship('transactions', 'customers')
+    s_to_r = get_relationship('stores', 'regions')
+    t_to_s = get_relationship('transactions', 'stores')
+
+    amount = ft.IdentityFeature(es['transactions']['amount'])
+    through_customers = ft.AggregationFeature(amount, es['regions'],
+                                              primitive=ft.primitives.Sum,
+                                              relationship_path=[c_to_r, t_to_c])
+    through_stores = ft.AggregationFeature(amount, es['regions'],
+                                           primitive=ft.primitives.Sum,
+                                           relationship_path=[s_to_r, t_to_s])
+
+    pandas_backend = PandasBackend(es, [through_customers, through_stores])
+    df = pandas_backend.calculate_all_features(instance_ids=[0, 1, 2],
+                                               time_last=datetime(2011, 4, 8))
+    assert (df['SUM(stores.transactions.amount)'] == [94, 261, 128]).all()
+    assert (df['SUM(customers.transactions.amount)'] == [72, 411, 0]).all()
+
+
+def test_two_relationships_to_single_entity(games_es):
+    es = games_es
+    home_team, away_team = es.relationships
+    mean_at_home = ft.AggregationFeature(es['games']['home_team_score'],
+                                         es['teams'],
+                                         relationship_path=[home_team],
+                                         primitive=ft.primitives.Mean)
+    mean_at_away = ft.AggregationFeature(es['games']['away_team_score'],
+                                         es['teams'],
+                                         relationship_path=[away_team],
+                                         primitive=ft.primitives.Mean)
+    home_team_mean = ft.DirectFeature(mean_at_home, es['games'],
+                                      relationship=home_team)
+    away_team_mean = ft.DirectFeature(mean_at_away, es['games'],
+                                      relationship=away_team)
+
+    pandas_backend = PandasBackend(es, [home_team_mean, away_team_mean])
+    df = pandas_backend.calculate_all_features(instance_ids=range(3),
+                                               time_last=datetime(2011, 8, 28))
+    assert (df[home_team_mean.get_name()] == [1.5, 1.5, 2.5]).all()
+    assert (df[away_team_mean.get_name()] == [1, 0.5, 2]).all()
+
+
+def test_entityset_with_self_loop(self_loop_es):
```

Comment:
I have left this failing test to prompt discussion of whether we want to support this case. This currently fails because we do not detect that paths with cycles are not unique. This means that two features with different length paths through the same cycle will be given the same name, and will be treated as the same feature in sets and dictionaries. In this case "COUNT(employees.employees)", cannot be distinguished from "COUNT(employees)".

The root cause is that in #544 we decided to "simplify" feature names when the path is unique ("MEAN(log.value)" instead of "MEAN(sessions.log.value)"), but to implement this we reused the existing methods for finding paths, which ignore cycles (to avoid infinite loops). If we want to support entitysets with cycles we could implement specific logic for detecting them, or decide that "simplifying" feature names is not necessary.

Reply:
As discussed in person, we are not going to add support for entitysets with loops in this PR, so I have removed this test.